include/xtensor-sparse/xmap.hpp

```diff
@@ -0,0 +1,285 @@
+#ifndef XTENSOR_SPARSE_LIL_HPP
+#define XTENSOR_SPARSE_LIL_HPP
+
+#include <array>
+#include <map>
+#include <utility>
+#include <vector>
+
+#include <xtensor/xstorage.hpp>
+#include <xtensor/xexception.hpp>
+#include <xtensor/xtensor.hpp>
+#include <xtensor/xadapt.hpp>
+#include <xtensor/xio.hpp>
+
+namespace xt
+{
+    template<class T>
+    class xmap_container
+    {
+    public:
+
+        using value_type = T;
+        using const_reference = const T&;
+        using reference = T&;
+        using pointer = T*;
+        using const_pointer = const T*;
+        using size_type = std::size_t;
+        using difference_type = std::ptrdiff_t;
+
+        using shape_type = svector<size_type>;
+        using strides_type = svector<size_type>;
+        using backstrides_type = svector<size_type>;
```

Comment:
`strides_type` and `backstrides_type` should not be in the public API

Reply:
we don't need it anymore. I forget to remove these types.